readthedocs/api/v3/views.py

```diff
@@ -126,6 +130,19 @@ class ProjectsViewSet(APIv3Settings, NestedViewSetMixin, ProjectQuerySetMixin,
         'active_versions.last_build.config',
     ]
 
+    def get_serializer_class(self):
+        """
+        Return correct serializer depending on the action.
+
+        For GET it returns a serializer with many fields and on PUT/PATCH/POST,
+        it return a serializer to validate just a few fields.
+        """
+        if self.action in ('list', 'retrieve', 'superproject'):
+            return ProjectSerializer
+
+        if self.action in ('create',):
```

Comment:
any reason not to just use a list here? Or check for the value explicitly?

Reply:
Just for consistency.